Warnings with -Wextra in CUDA part

I am getting tons of warnings with Thrust v1.8.3-2 (bundled with CUDA 8.0) when compiling CUDA-related code with `-Wextra`. Most of them are `-Wunsued-parameter` warnings and there are lots of duplicates blowing up my console. So I think that there are not so many of them but they get triggered quite easily.

I am very interested in fixing them since I want to enable `-Wextra` in my project for both host and device code. Let me know if I should hand in a pull request for that.

Thanks @RealRecon, I see the warnings now (below). The specific culprit seems to be `-Wunused-parameter`, not necessarily `-Wextra`.


```
$ nvcc reduce.cu -Xcompiler -Wunused-parameter
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/execute_on_stream.h: In instantiation of ‘CUstream_st* thrust::system::cuda::detail::stream(const thrust::system::cuda::detail::execution_policy<System1>&) [with DerivedPolicy = thrust::system::cuda::detail::tag; cudaStream_t = CUstream_st*]’:
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:106:24:   required from ‘OutputType thrust::system::cuda::detail::reduce_detail::tuned_reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:202:35:   required from ‘typename thrust::detail::enable_if<thrust::detail::is_arithmetic<OutputType>::value, OutputType>::type thrust::system::cuda::detail::reduce_detail::reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>; typename thrust::detail::enable_if<thrust::detail::is_arithmetic<OutputType>::value, OutputType>::type = int]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:257:29:   required from ‘static OutputType thrust::system::cuda::detail::reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction)::workaround::parallel_path(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:269:1:   required from ‘OutputType thrust::system::cuda::detail::reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:71:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/system/detail/generic/reduce.inl:52:22:   required from ‘T thrust::system::detail::generic::reduce(thrust::execution_policy<Derived>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:54:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:168:22:   required from ‘T thrust::reduce(InputIterator, InputIterator, T) [with InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
reduce.cu:10:55:   required from here
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/execute_on_stream.h:55:48: warning: unused parameter ‘exec’ [-Wunused-parameter]
 inline cudaStream_t stream(const execution_policy<DerivedPolicy> &exec)
                                                ^
/usr/local/cuda/bin/..//include/thrust/detail/allocator/temporary_allocator.inl: In instantiation of ‘void thrust::detail::temporary_allocator<T, System>::deallocate(thrust::detail::temporary_allocator<T, System>::pointer, thrust::detail::temporary_allocator<T, System>::size_type) [with T = int; System = thrust::system::cuda::detail::tag; thrust::detail::temporary_allocator<T, System>::pointer = thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default>; thrust::detail::temporary_allocator<T, System>::size_type = long unsigned int]’:
/usr/local/cuda/bin/..//include/thrust/detail/allocator/temporary_allocator.inl:46:11:   required from ‘thrust::detail::temporary_allocator<T, System>::pointer thrust::detail::temporary_allocator<T, System>::allocate(thrust::detail::temporary_allocator<T, System>::size_type) [with T = int; System = thrust::system::cuda::detail::tag; thrust::detail::temporary_allocator<T, System>::pointer = thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default>; thrust::detail::temporary_allocator<T, System>::size_type = long unsigned int]’
/usr/local/cuda/bin/..//include/thrust/detail/allocator/allocator_traits.inl:229:22:   required from ‘static thrust::detail::allocator_traits<Alloc>::pointer thrust::detail::allocator_traits<Alloc>::allocate(Alloc&, thrust::detail::allocator_traits<Alloc>::size_type)::workaround_warnings::allocate(Alloc&, thrust::detail::allocator_traits<Alloc>::size_type) [with Alloc = thrust::detail::no_throw_allocator<thrust::detail::temporary_allocator<int, thrust::system::cuda::detail::tag> >; thrust::detail::allocator_traits<Alloc>::pointer = thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default>; thrust::detail::allocator_traits<Alloc>::size_type = long unsigned int]’
/usr/local/cuda/bin/..//include/thrust/detail/allocator/allocator_traits.inl:231:1:   required from ‘static thrust::detail::allocator_traits<Alloc>::pointer thrust::detail::allocator_traits<Alloc>::allocate(Alloc&, thrust::detail::allocator_traits<Alloc>::size_type) [with Alloc = thrust::detail::no_throw_allocator<thrust::detail::temporary_allocator<int, thrust::system::cuda::detail::tag> >; thrust::detail::allocator_traits<Alloc>::pointer = thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default>; thrust::detail::allocator_traits<Alloc>::size_type = long unsigned int]’
/usr/local/cuda/bin/..//include/thrust/detail/contiguous_storage.inl:155:47:   required from ‘void thrust::detail::contiguous_storage<T, Alloc>::allocate(thrust::detail::contiguous_storage<T, Alloc>::size_type) [with T = int; Alloc = thrust::detail::no_throw_allocator<thrust::detail::temporary_allocator<int, thrust::system::cuda::detail::tag> >; thrust::detail::contiguous_storage<T, Alloc>::size_type = long unsigned int]’
/usr/local/cuda/bin/..//include/thrust/detail/contiguous_storage.inl:55:9:   required from ‘thrust::detail::contiguous_storage<T, Alloc>::contiguous_storage(thrust::detail::contiguous_storage<T, Alloc>::size_type, const Alloc&) [with T = int; Alloc = thrust::detail::no_throw_allocator<thrust::detail::temporary_allocator<int, thrust::system::cuda::detail::tag> >; thrust::detail::contiguous_storage<T, Alloc>::size_type = long unsigned int]’
/usr/local/cuda/bin/..//include/thrust/detail/temporary_array.inl:74:168:   [ skipping 4 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:269:1:   required from ‘OutputType thrust::system::cuda::detail::reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:71:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/system/detail/generic/reduce.inl:52:22:   required from ‘T thrust::system::detail::generic::reduce(thrust::execution_policy<Derived>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:54:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:168:22:   required from ‘T thrust::reduce(InputIterator, InputIterator, T) [with InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
reduce.cu:10:55:   required from here
/usr/local/cuda/bin/..//include/thrust/detail/allocator/temporary_allocator.inl:62:67: warning: unused parameter ‘n’ [-Wunused-parameter]
     ::deallocate(typename temporary_allocator<T,System>::pointer p, typename temporary_allocator<T,System>::size_type n)
                                                                   ^
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/cuda_task.hpp: In instantiation of ‘static Block thrust::system::cuda::detail::bulk_::detail::block_maker<Block>::make(typename Block::size_type, typename Block::size_type, typename Block::agent_type, typename Block::size_type) [with Block = thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>; typename Block::size_type = int; typename Block::agent_type = thrust::system::cuda::detail::bulk_::agent<7ul>]’:
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/cuda_task.hpp:196:34:   required from ‘Block thrust::system::cuda::detail::bulk_::detail::make_block(typename Block::size_type, typename Block::size_type, typename Block::agent_type, typename Block::size_type) [with Block = thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>; typename Block::size_type = int; typename Block::agent_type = thrust::system::cuda::detail::bulk_::agent<7ul>]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/cuda_launcher/cuda_launcher.hpp:293:67:   required from ‘thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::grid_type thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::configure(thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::grid_type) [with long unsigned int gridsize = 0ul; long unsigned int blocksize = 128ul; long unsigned int grainsize = 7ul; Closure = thrust::system::cuda::detail::bulk_::detail::closure<thrust::system::cuda::detail::reduce_detail::reduce_partitions, thrust::tuple<thrust::system::cuda::detail::bulk_::detail::cursor<1u>, thrust::detail::normal_iterator<thrust::device_ptr<int> >, thrust::system::cuda::detail::aligned_decomposition<long int>, thrust::detail::normal_iterator<thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default> >, int, thrust::plus<int>, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type> >; thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::grid_type = thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>, 0ul>]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/cuda_launcher/cuda_launcher.hpp:253:24:   required from ‘void thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::launch(thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::grid_type, Closure, cudaStream_t) [with long unsigned int gridsize = 0ul; long unsigned int blocksize = 128ul; long unsigned int grainsize = 7ul; Closure = thrust::system::cuda::detail::bulk_::detail::closure<thrust::system::cuda::detail::reduce_detail::reduce_partitions, thrust::tuple<thrust::system::cuda::detail::bulk_::detail::cursor<1u>, thrust::detail::normal_iterator<thrust::device_ptr<int> >, thrust::system::cuda::detail::aligned_decomposition<long int>, thrust::detail::normal_iterator<thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default> >, int, thrust::plus<int>, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type> >; thrust::system::cuda::detail::bulk_::detail::cuda_launcher<thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<grainsize>, blocksize>, gridsize>, Closure>::grid_type = thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>, 0ul>; cudaStream_t = CUstream_st*]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/async.inl:46:1:   required from ‘thrust::system::cuda::detail::bulk_::future<void> thrust::system::cuda::detail::bulk_::detail::async_in_stream(ExecutionGroup, Closure, cudaStream_t, cudaEvent_t) [with ExecutionGroup = thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>, 0ul>; Closure = thrust::system::cuda::detail::bulk_::detail::closure<thrust::system::cuda::detail::reduce_detail::reduce_partitions, thrust::tuple<thrust::system::cuda::detail::bulk_::detail::cursor<1u>, thrust::detail::normal_iterator<thrust::device_ptr<int> >, thrust::system::cuda::detail::aligned_decomposition<long int>, thrust::detail::normal_iterator<thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default> >, int, thrust::plus<int>, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type> >; cudaStream_t = CUstream_st*; cudaEvent_t = CUevent_st*]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/async.inl:96:62:   required from ‘thrust::system::cuda::detail::bulk_::future<void> thrust::system::cuda::detail::bulk_::detail::async(thrust::system::cuda::detail::bulk_::async_launch<ExecutionGroup>, Closure) [with ExecutionGroup = thrust::system::cuda::detail::bulk_::parallel_group<thrust::system::cuda::detail::bulk_::concurrent_group<thrust::system::cuda::detail::bulk_::agent<7ul>, 128ul>, 0ul>; Closure = thrust::system::cuda::detail::bulk_::detail::closure<thrust::system::cuda::detail::reduce_detail::reduce_partitions, thrust::tuple<thrust::system::cuda::detail::bulk_::detail::cursor<1u>, thrust::detail::normal_iterator<thrust::device_ptr<int> >, thrust::system::cuda::detail::aligned_decomposition<long int>, thrust::detail::normal_iterator<thrust::pointer<int, thrust::system::cuda::detail::tag, thrust::use_default, thrust::use_default> >, int, thrust::plus<int>, thrust::null_type, thrust::null_type, thrust::null_type, thrust::null_type> >]’
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/async.inl:157:21:   [ skipping 4 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/reduce.inl:269:1:   required from ‘OutputType thrust::system::cuda::detail::reduce(thrust::system::cuda::detail::execution_policy<System1>&, InputIterator, InputIterator, OutputType, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; OutputType = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:71:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T, BinaryFunction) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int; BinaryFunction = thrust::plus<int>]’
/usr/local/cuda/bin/..//include/thrust/system/detail/generic/reduce.inl:52:22:   required from ‘T thrust::system::detail::generic::reduce(thrust::execution_policy<Derived>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:54:14:   required from ‘T thrust::reduce(const thrust::detail::execution_policy_base<DerivedPolicy>&, InputIterator, InputIterator, T) [with DerivedPolicy = thrust::system::cuda::detail::tag; InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
/usr/local/cuda/bin/..//include/thrust/detail/reduce.inl:168:22:   required from ‘T thrust::reduce(InputIterator, InputIterator, T) [with InputIterator = thrust::detail::normal_iterator<thrust::device_ptr<int> >; T = int]’
reduce.cu:10:55:   required from here
/usr/local/cuda/bin/..//include/thrust/system/cuda/detail/bulk/detail/cuda_task.hpp:161:45: warning: unused parameter ‘size’ [-Wunused-parameter]
   static Block make(typename Block::size_type     size,
```

--- a/thrust/detail/allocator/temporary_allocator.inl
+++ b/thrust/detail/allocator/temporary_allocator.inl
@@ -59,7 +59,7 @@ __host__ __device__
 template<typename T, typename System>
 __host__ __device__
   void temporary_allocator<T,System>
-    ::deallocate(typename temporary_allocator<T,System>::pointer p, typename temporary_allocator<T,System>::size_type n)
+    ::deallocate(typename temporary_allocator<T,System>::pointer p, typename temporary_allocator<T,System>::size_type)
 {
   return thrust::return_temporary_buffer(system(), p);
 } // end temporary_allocator

--- a/thrust/system/cuda/detail/bulk/detail/cuda_task.hpp
+++ b/thrust/system/cuda/detail/bulk/detail/cuda_task.hpp
@@ -158,7 +158,7 @@ template<typename Block>
 struct block_maker
 {
   __host__ __device__
-  static Block make(typename Block::size_type     size,
+  static Block make(typename Block::size_type,
                     typename Block::size_type     heap_size,
                     typename Block::agent_type    thread,
                     typename Block::size_type     index)

--- a/thrust/system/cuda/detail/cub/util_allocator.cuh
+++ b/thrust/system/cuda/detail/cub/util_allocator.cuh
@@ -270,7 +270,8 @@ struct CachingDeviceAllocator
             min_bin_bytes(IntPow(bin_growth, min_bin)),
             max_bin_bytes(IntPow(bin_growth, max_bin)),
             max_cached_bytes(max_cached_bytes),
-            debug(false)
+            debug(false),
+            skip_cleanup(skip_cleanup)
     #if (CUB_PTX_ARCH == 0)   // Only define STL container members in host code
             ,cached_blocks(BlockDescriptor::SizeCompare),
             live_blocks(BlockDescriptor::PtrCompare)

--- a/thrust/system/cuda/detail/cub/util_debug.cuh
+++ b/thrust/system/cuda/detail/cub/util_debug.cuh
@@ -70,6 +70,9 @@ __host__ __device__ __forceinline__ cudaError_t Debug(
     const char*     filename,
     int             line)
 {
+    // silence unused parameter warnings
+    (void)filename;
+    (void)line;
 #ifdef CUB_STDERR
     if (error)
     {

--- a/thrust/system/cuda/detail/cub/util_type.cuh
+++ b/thrust/system/cuda/detail/cub/util_type.cuh
@@ -122,11 +122,11 @@ struct NullType
 #ifndef DOXYGEN_SHOULD_SKIP_THIS    // Do not document
 
     template <typename T>
-    __host__ __device__ __forceinline__ NullType& operator =(const T& b) { return *this; }
+    __host__ __device__ __forceinline__ NullType& operator =(const T&) { return *this; }
 
-    __host__ __device__ __forceinline__ bool operator ==(const NullType& b) { return true; }
+    __host__ __device__ __forceinline__ bool operator ==(const NullType&) { return true; }
 
-    __host__ __device__ __forceinline__ bool operator !=(const NullType& b) { return false; }
+    __host__ __device__ __forceinline__ bool operator !=(const NullType&) { return false; }
 
 #endif // DOXYGEN_SHOULD_SKIP_THIS
 };

--- a/thrust/system/cuda/detail/execute_on_stream.h
+++ b/thrust/system/cuda/detail/execute_on_stream.h
@@ -52,7 +52,7 @@ inline cudaStream_t default_stream()
 // given any old execution_policy, we return the default stream
 template<typename DerivedPolicy>
 __host__ __device__
-inline cudaStream_t stream(const execution_policy<DerivedPolicy> &exec)
+inline cudaStream_t stream(const execution_policy<DerivedPolicy> &)
 {
   return default_stream();
 }

